exec/cnex/disassembly.c

```diff
@@ -0,0 +1,855 @@
+#include "disassembly.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#include "bytecode.h"
+#include "exec.h"
+#include "module.h"
+#include "nstring.h"
+#include "opcode.h"
+
+#ifdef _MSC_VER
+#define snprintf _snprintf
+#endif
+
+
+static TInstructionDisassembler *g_disasm;
+
+TInstructionDisassembler *disasm_createDisassembler(void)
+{
+    TInstructionDisassembler *r = malloc(sizeof(TInstructionDisassembler));
+    if (r == NULL) {
+        fatal_error("Could not allocate memory for Instruction Dissembler.");
+    }
+    memset(r->out, 0x00, sizeof(r->out));
+    return r;
+}
+
+char *disasm_disassembleInstruction(TExecutor *exec)
+{
+    if (g_disasm == NULL) {
```

Comment:
I don't understand why this needs to be a global. Why not make it a static inside this function? Then you wouldn't need the whole dynamic allocation in `disasm_createDisassembler` either. This seems a lot like making an unnecessary object out of something that only needs to be a function.

Reply:
Created this as a static, and removed all of the dynamic allocation.  Originally, I was thinking of allowing trace output to go to other devices, like files, or sockets.  But, with shell redirection that is still possible; plus I thought I might had needed to do more with the Disassembly object, but it turned out to be easier than I imagined.